exec/cnex/support.c

```diff
@@ -0,0 +1,181 @@
+#ifdef _WIN32
+#define _CRT_SECURE_NO_WARNINGS
+#endif
+
+#include <assert.h>
+#include <iso646.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "module.h"
+#include "support.h"
+#include "util.h"
+
+#ifdef _WIN32
+#include <io.h>
+#define strdup _strdup
+#define PATHSEP ';'
+#else
+#include <unistd.h>
+#define PATHSEP ':'
+#endif
+
+typedef struct tagTPath {
+    char *path;
+    struct tagTPath *next;
+} TPath;
+
+// We only need to declare this once, in one location.
+static TPath *neon_paths;
+
+static void path_pushPath(const char *source_path)
+{
+    // Always start at the root...
+    TPath *p = neon_paths;
+
+    TPath *n = malloc(sizeof(TPath));
+    n->path = strdup(source_path);
+    n->next = NULL;
+
+    // If we haven't initialized the Neon paths yet, go ahead and set n to the root and return.
+    if (p == NULL) {
+        neon_paths = n;
+        return;
+    }
+
+    // Jump to the end of the list; to add this path onto our search paths.
+    while (p && p->next) {
+        p = p->next;
+    }
+
+    // Add this new path to the list of paths.
+    p->next = n;
+}
+
+void path_freePaths()
+{
+    TPath *p = neon_paths;
+    while (p) {
+        free(p->path);
+        TPath *n = p->next;
+        free(p);
+        p = n;
+    }
+}
+
+void path_initPaths(const char *source_path)
+{
+    // Paths are arlready initialized, so just return.
+    if (neon_paths) {
+        return;
+    }
+
+    // Push neon executable path first.
+    if (strlen(source_path)) {
+        path_pushPath(source_path);
+    }
+    // Then push the cwd path.
+    path_pushPath("./");
+
+    // Get paths from %NEONPATH% environment variable, if there is one.
+    char *ne = getenv("NEONPATH");
+    if (ne != NULL) {
+        char *next;
+        char *path = ne;
+        for (;;) {
+            next = strchr(path, PATHSEP);
+            if (next == NULL) {
+                break;
+            }
+            *next = '\0';
+            path_pushPath(path);
+            path = next + 1;
+        }
+        // Push the last (or perhaps only) path in the list to the list of paths, if it's not blank.
+        if (path != NULL && *path != '\0') {
+            path_pushPath(path);
+        }
+    }
+
+    // Next, check for .neonpath, and process any paths that might exist in it.
+    FILE *fp = fopen(".neonpath", "r");
+    if (fp) {
+        for (;;) {
+            char path[1024];
+            if (fgets(path, sizeof(path), fp) == NULL) {
+                break;
+            }
+            // ToDo: Trim possible leading and trailing spaces (whitespace) off loaded paths.
+            size_t len = strlen(path);
+            if (path[len-1] == '\n') {
```

Comment:
This doesn't do what the comment above says it does. This only trims a single trailing newline, while the comment says it will trim any leading or trailing whitespace.

Reply:
That's left as a ToDo.